docs/src/use/configure/migration-guide.md

```diff
@@ -0,0 +1,422 @@
+
+* [`.eslintrc` configuration files](configuration-files)
+* [`eslint.config.js` configuration files](configuration-files-new)
+
+## Start Using `eslint.config.js`
+
+Starting with ESLint v9.0.0, the `eslint.config.js` file format will be the default configuration file format. Once ESLint v9.0.0 is released, you can start using the `eslint.config.js` file format without any additional configuration.
+
+To use `eslint.config.js` with ESLint v8, place a `eslint.config.js` file in the root of your project **or** set an `ESLINT_USE_FLAT_CONFIG` environment variable to `true`.
+
+## Key Differences between Configuration Formats
+
+A few of the most notable differences between the `.eslintrc` and `eslint.config.js` formats are the following:
+
+### Importing Plugins and Custom Parsers
+
+#### .eslintrc
+
+String-based import system with the `plugins` property to load plugins and `extends` to load configurations from the plugins.
+
+```javascript
+// .eslintrc.js
+
+{
+    // ...other config
+    plugins: ["jsdoc"],
+    rules: {
+        "jsdoc/require-description": "error",
+        "jsdoc/check-values": "error"
+    }
+    // ...other config
+}
+```
+
+#### eslint.config.js
+
+Uses CommonJS `requires()` or ESModule `import` to load plugins and custom parsers.
+
+```javascript
+// eslint.config.js
+
+import jsdoc from "eslint-plugin-jsdoc";
+
+export default [
+    {
+        files: ["**/*.js"],
+        plugins: {
+            jsdoc: jsdoc
+        },
+        rules: {
+            "jsdoc/require-description": "error",
+            "jsdoc/check-values": "error"
+        }
+    }
+];
+```
+
+### Custom Parsers
+
+#### .eslintrc
+
+Importing a custom parser is very similar to importing a plugin. However, a parser must be a separate module.
+
+```javascript
+// .eslintrc.js
+
+{
+    // ...other config
+    parser: "@babel/eslint-parser",
+    // ...other config
+}
+```
+
+#### eslint.config.js
+
+To import a custom parser, import the parser as a module. Then assign it to the `languageOptions.parser` property of a configuration object:
+
+```javascript
+// eslint.config.js
+
+import babelParser from "@babel/eslint-parser";
+
+export default [
+   {
+       // ...other config
+       languageOptions: {
+           parser: babelParser
+       }
+       // ...other config
+   }
+];
+```
+
+### Glob-Based Configs
+
+#### .eslintrc
+
+By default, .`eslintrc` files lint all files (except those covered by `.gitignore`) in the directory in which they’re placed and its child directories:
+
+```javascript
+// .eslintrc.js
+
+module.exports = {
+    // ...other config
+    rules: {
+        semi: ["warn", "always"]
+    }
+};
+```
+
+If you want to have different configurations for different file glob patterns, you can specify them in the `overrides` property:
+
+```javascript
+// .eslintrc.js
+
+module.exports = {
+    // ...other config
+    extends: "eslint:recommended",
```

Comment:
I'd remove this line as `extends` isn't in the previous config and doesn't really factor into the discussion.
```suggestion
```